### ISSUE TITLE
$form->getElements() is empty

### ISSUE THREAD

**Issue by BrunoSpy** [laminas/laminas-form#95]
### BC Break Report

|    Q        |   A
|------------ | ------
| Version     | 2.16.1

#### Summary

```php
$category = new Category();
$builder = new AnnotationBuilder();
$form = $builder->createForm(Category::class);
$form->setHydrator(new DoctrineObject($objectManager))->setObject($category);
```
#### Previous behavior

$form->getElements() is an array with all expected elements

#### Current behavior

$form->getElements() is empty

#### How to reproduce

Annotations are visible here : https://gitlab.asap.dsna.fr/bruno.spyckerelle/epeires2/-/blob/test/module/Application/src/Entity/Category.php

**Comment by Slamdunk** [comment]
I see you used [laminas-code:3.4.1](https://gitlab.asap.dsna.fr/bruno.spyckerelle/epeires2/-/blob/91e548ff27c41ad2467dd506db6fe83138abfcc2/composer.lock#L4923-4924) and forcing platform PHP 7.3 even on PHP 8.0 builds.

You'd need `laminas-code:3.5.0` for PHP 8 build, can you try please? If the build succeeds, I'll release a `2.16.2` version which conflicts with `laminas-code < 3.5.0`

**Comment by BrunoSpy** [comment]
I confirm that forcing laminas-code to ^2.5.0 fixes the bug but why laminas-form 2.15 uses laminas-code 3.5 and 2.16 uses 3.4 ?
